### ISSUE TITLE
AerospikeTemplate.update operation does not consider document version

### ISSUE THREAD

**Issue by Aloren** [aerospike-community/spring-data-aerospike#51]
See more details in #47.

**TL;DR**
Currently `update` operation in `AerospikeTemplate` does not consider document's version property and overwrites document even when document's version is not equal to server version.

**Thoughts**
If we change this behavior -- we can break existing users that expect such behavior.
If we do not change this behavior -- some users may potentially overwrite data that has not been seen.
Also we do not know whether such behavior in `spring-data-aerospike` for update operation was intended or not. Probably @helipilot50 could help with that as the first `update` implementation was done by him.

FYI @vasilievip @tdanylchuk

**Comment by Aloren** [comment]
Discussed internally that the issue should be fixed. 
In case somebody needs this specific behavior -- we need to discuss such use cases.

**Comment by Aloren** [comment]
Fixed in #55
